processor/groupbytraceprocessor/processor_test.go

```diff
@@ -41,6 +43,7 @@ func TestTraceIsDispatchedAfterDuration(t *testing.T) {
 	config := Config{
 		WaitDuration: time.Nanosecond,
 		NumTraces:    10,
+		NumWorkers:   1,
```

Comment:
I'm a bit worried that we might not have enough tests with NumWorkers higher than 1, so that possible race conditions are not being detected.